ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -1847,6 +1844,14 @@ FEDataManager::interpWeighted(const int f_data_idx,
     // Accumulate data.
     if (close_F) F_vec.close();
 
+    // Enforce constraints here if using nodal quadrature.  (If using
+    // element-based quadrature, constraints have already been imposed above.)
+    if (use_nodal_quadrature && F_dof_map.n_constrained_dofs() > 0)
+    {
+        if (!close_F) F_vec.close();
+        F_dof_map.enforce_constraints_exactly(F_system, &F_vec);
```

Comment:
This doesn't seem right - here F_vec is a RHS vector, not a solution vector. We should be performing assembly in the same way in both the nodal and nonnodal case.

Reply:
No --- nodal interpolation is just direct interpolation at the nodes --- it is nothing like the elemental case. We sort of fake it to be like the elemental case by multiplying by the diagonal mass matrix.

Reply:
Maybe we need to impose the constraints on the assembled interpolated vector and then multiply by the diagonal mass matrix.